src/main/java/wanya/task/TaskList.java

```diff
@@ -0,0 +1,219 @@
+        Task newTask = new Deadline(taskName, dueDate);
+        return addTask(newTask);
+    }
+
+    /**
+     * Adds an Event task to the TaskList.
+     *
+     * @param commandInput user input with command event.
+     * @throws WanyaException if input does not contain /at and date.
+     * @throws DateTimeException if invalid date time format is provided.
+     * @return String message for adding an Event task.
+     */
+    public String addEvent(String commandInput) throws WanyaException, DateTimeException {
+        String[] inputs = commandInput.split("/at");
+        //no date provided
+        if (inputs.length == 1) {
+            throw new WanyaException("Event must have a date\n" +
+                    "Include '/at' and date with format " +
+                    "\"yyyy-mm-dd\" at the back");
+        }
+        String taskName = inputs[0];
+        String date = inputs[1].trim();
+        Task newTask = new Event(taskName, date);
+        return addTask(newTask);
+    }
+
+    /**
+     * Adds the task to the TaskList and display task added.
+     *
+     * @param task task to be added.
+     * @return String message for adding a task.
+     */
+    private String addTask(Task task) {
+        tasks.add(task);
+        return("You have added: \n"
+                + task
+                + "\nNow you have " + size() + " tasks in your list.");
+    }
+
+    /**
+     * Deletes a task from the TaskList.
+     *
+     * @param index the task number to be deleted.
+     * @return String message for successfully deleting a task.
+     */
+    public String deleteTask(int index) {
+        Task task = tasks.get(index - 1);
+        tasks.remove(index - 1);
+        return("WWaku WWaku!!! Wanya has used her magic powers to remove this task:\n"
+                + task
+                + "\nNow you have " + size() + " tasks in your list.");
+    }
+
+    /**
+     * Displays the list of tasks in TaskList.
+     *
+     * @return List of tasks in String.
+     */
+    public String showTasks() {
+        if (isEmpty()) {
+            return("List is empty! Wheee slack time!");
+        }
+
+        StringBuilder taskString = new StringBuilder();
+        taskString.append("Your current list now!\n");
+        for (int i = 1; i <= size(); i++) {
+            Task task = tasks.get(i - 1);
+            taskString.append(i + "." + task + "\n");
+        }
+        return taskString.toString();
+    }
+
+    /**
+     * Encodes the list of tasks in TaskList.
+     *
+     * @return List of Strings of the tasks in TaskList to be stored.
+     */
+    public List<String> saveToStorage() {
+        List<String> data = new ArrayList<>();
+        for (int i = 0; i < size(); i++) {
+            Task task = tasks.get(i);
+            data.add(task.toStorageString());
+        }
+        return data;
+    }
+
+    /**
+     * Finds a list of tasks that contains the keyword and print it.
+     *
+     * @param keyword word to find matching tasks.
```

Comment:
Perhaps you would like to capitalise the first letter in the word "word"? As I believe that this may violate JavaDoc standards.